syntax = "proto3";


service UserService {
  rpc Register (RegisterRequest) returns (UserResponse);
  rpc Login (LoginRequest) returns (LoginResponse);
}

message RegisterRequest {
  string email = 1;
  string username = 2;
  string password = 3;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message UserResponse {
  string message = 1;
  bool success = 2;         // ← أضف هذا
}

message LoginResponse {
  string message = 1;
  bool success = 2;         // ← أضف هذا
  int32 user_id = 3;        // ← إذا كنت ترجع ID بعد تسجيل الدخول
}
